File: gaia-sdk-python/gaia_sdk/graphql/request/type/UpdatedPromptImpulse.py

from gaia_sdk.graphql.request.type.Prompt import Prompt

from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry


class UpdatedPromptImpulse(list):
    """
    Impulse which indicates the resulf of a update prompt impulse
    """

    def id(self):
        self.append(lambda x: "id")

    """
    the prompt instance
    """
    def prompt(self, config: Callable[['Prompt'], None]):
        def callback(registry: VariableRegistry):
            entity = Prompt()
            config(entity)
            return "prompt {" + entity.render(registry) + "}"
        self.append(callback)

    def render(self, registry: VariableRegistry):
        return " ".join(map(lambda e: e(registry), self))

File: gaia-sdk-javascript/src/graphql/request/type/UpdatedPromptImpulse.ts

import {Prompt} from "./Prompt";

import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * Impulse which indicates the result of a update prompt impulse
 */
export class UpdatedPromptImpulse extends Array<(_:VariableRegistry) => string> {
public _typeName = "UpdatedPromptImpulse";
    public id = () => { 
        this.push(_ => "id")
    };

    /**
     * the prompt instance
     */
    public data = (config: (_:Prompt) => void) => this.push((registry) => {
        const entity = new Prompt();
        config(entity);
        return "data { " + entity.render(registry) + " }";
    });

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

